client/src/components/game/gui/meters/Meters.jsx

```diff
@@ -44,7 +46,26 @@ function Meters() {
     const [energy, setEnergy] = useState(0);
     const [maxEnergy, setMaxEnergy] = useState(0);
 
+    const [combatTimer, setCombatTimer] = useState(0);
+
+    let combatTimerUpdateInterval = -1;
+    let combatTimerInternal = 0;
+
     useEffect(() => {
+        const updateCombatTimer = () => {
+            combatTimerInternal = Math.max(0, combatTimerInternal - 1000);
+            setCombatTimer(combatTimerInternal);
+        };
+
+        const refreshCombatTimer = () => {
+            if (combatTimerUpdateInterval !== -1) {
+                clearInterval(combatTimerUpdateInterval);
+            }
+            combatTimerUpdateInterval = setInterval(() => updateCombatTimer(), 1000);
+        };
```

Comment:
I think these functions can be declared above the useEffect, in the same scope as the `let combatTimer...`